src/NUnitFramework/framework/Attributes/TypeGetMemberExtension.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace NUnit.Framework
+{
+    internal static class TypeGetMemberExtension
+    {
+        /// <summary>
+        /// Searches for the specified members, using the specified binding constraints.
+        /// </summary>
+        /// <remarks>
+        /// Similar to <see cref="Type.GetMember(string, BindingFlags)"/> but also finds private static members from the base class.
+        /// </remarks>
+        /// <param name="type">The type to find the members from.</param>
+        /// <param name="name">The string containing the name of the members to get.</param>
+        /// <param name="flags">A bitwise combination of the enumeration values that specify how the search is conducted.</param>
+        /// <returns></returns>
+        public static MemberInfo[] GetMemberIncludingFromBase(this Type type, string name, BindingFlags flags)
+        {
+            var members = new List<MemberInfo>();
+            do
+            {
+                members.AddRange(type.GetMember(name, flags));
+            }
+            while (members.Count == 0 && (type = type.BaseType) != null);
```

Comment:
Suggestion:  `NUnit.Framework.Internal.Reflect` also defines a few extension methods over the `Type` type, including this "walking the class hierarchy" behaviour. Would it make sense to use [`TypeAndBaseTypes`](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Internal/Reflect.cs#L291-L297) here instead?

I know this part changed recently, but if I'm reading the current version of the code right, we also only walk the hierarchy until the first time that `GetMember` returns something. Coupled with the above suggestion, would it be simpler to change this from a dowhile to a simple foreach + return like the below pseudocode?

```
foreach currentType in type.TypeAndBaseTypes()
  members = type.GetMember(name, flags)
  if members.Count != 0
    return members
```